avm/ptn/sa/content-processing/main.bicep

```diff
@@ -0,0 +1,2183 @@
+//     roleDefinitionId: '7f951dda-4ed3-4680-a7ca-43fe172d538d' // AcrPull role
+//     principalType: 'ServicePrincipal'
+//     enableTelemetry: enableTelemetry
+//   }
+//   scope: resourceGroup(resourceGroup().name)
+// }
+
+// ========== Container App  ========== //
+module avmContainerApp 'br/public:avm/res/app/container-app:0.17.0' = {
+  name: format(resourceNameFormatString, 'caapp-')
+  params: {
+    name: 'ca-${solutionPrefix}-app'
+    location: resourceGroupLocation
+    environmentResourceId: avmContainerAppEnv.outputs.resourceId
+    workloadProfileName: 'Consumption'
+    enableTelemetry: enableTelemetry
+    registries: useLocalBuild == 'localbuild'
+      ? [
+          {
+            server: publicContainerImageEndpoint
+            identity: avmContainerRegistryReader.outputs.principalId
+          }
+        ]
+      : null
+
+    managedIdentities: {
+      systemAssigned: true
+      userAssignedResourceIds: [
+        avmContainerRegistryReader.outputs.resourceId
+      ]
+    }
+
+    containers: [
+      {
+        name: 'ca-${solutionPrefix}'
+        image: '${publicContainerImageEndpoint}/contentprocessor:latest'
+
+        resources: {
+          cpu: '4'
+          memory: '8.0Gi'
+        }
+        env: [
+          {
+            name: 'APP_CONFIG_ENDPOINT'
+            value: ''
+          }
+        ]
+      }
+    ]
+    activeRevisionsMode: 'Single'
+    ingressExternal: false
+    disableIngress: true
+    // scaleSettings: {
+    //   minReplicas: 1
+    //   maxReplicas: 1
+    // }
+    scaleSettings: {
+      maxReplicas: enableScaling ? 3 : 2
+      minReplicas: enableScaling ? 2 : 1
+    }
+    tags: tags
+  }
+}
+
+// ========== Container App API ========== //
+module avmContainerApp_API 'br/public:avm/res/app/container-app:0.17.0' = {
+  name: format(resourceNameFormatString, 'caapi-')
+  params: {
+    name: 'ca-${solutionPrefix}-api'
+    location: resourceGroupLocation
+    environmentResourceId: avmContainerAppEnv.outputs.resourceId
+    workloadProfileName: 'Consumption'
+    enableTelemetry: enableTelemetry
+    registries: useLocalBuild == 'localbuild'
+      ? [
+          {
+            server: avmContainerRegistry.outputs.loginServer
+            identity: avmContainerRegistryReader.outputs.principalId
+          }
+        ]
+      : null
+    // registries: useLocalBuild == 'localbuild'
+    //   ? [
+    //       {
+    //         server: publicContainerImageEndpoint
+    //         image: 'contentprocessorapi'
+    //         imageTag: 'latest'
+    //       }
+    //     ]
+    //   : null
+    tags: tags
+    managedIdentities: {
+      systemAssigned: true
+      userAssignedResourceIds: [
+        avmContainerRegistryReader.outputs.resourceId
+      ]
+    }
+
+    containers: [
+      {
+        name: 'ca-${solutionPrefix}-api'
+        image: (useLocalBuild != 'localbuild')
+          ? '${publicContainerImageEndpoint}/contentprocessorapi:latest'
```

Comment:
Hardcoding latest image is not a good idea, specially if at some point, there is a non backward compatible change that depends on the infrastructure (e.g. depends of a new environment variable).

Recommended to create a parameter with a default value set to a specific version of the image that works for the current - e.g. `param backendContainerImageTag string = 'latest_2025-06-17_704'`

See how we are addressing in [Multi Agent](https://github.com/Azure/bicep-registry-modules/blob/main/avm/ptn/sa/multi-agent-custom-automation-engine/main.bicep#L52)